### ISSUE TITLE
Catch unary operator after keyword await as error

### ISSUE THREAD

**Issue by timnew** [babel/babel-eslint#203]
To invert the boolean value (or any unary operator, such as `~` or `-` for numbers) returned from promise with `await`, we should put the operator before the `await` keyword rather than after. But this is counterintuitive and error-prone.

e.g.
```js
function returnFalse() {
  return Promise.resolve(false);
}

async function test() {
  if(await !returnFalse()) {
    console.log('ok');
  }  
}

test();
```

We might expect `ok` to be displayed, but it never does. Since we apply `!` on the `promise` return by async function, which is meaningless in most cases. So it would be nice if any code that apply `!` after `await` rather than before.

Here is the correct `test()`:
```js
async function test() {
  if(! await returnFalse()) {
    console.log('ok');
  }  
}

```

I think it will be a general issue when writing code with `async/await`. And I think linter should check fo r this case.

**Comment by hzoo** [comment]
This sounds like a new rule to be proposed - not an issue with babel-eslint itself. I would suggest it in our corresponding repo: https://github.com/babel/eslint-plugin-babel?

**Comment by timnew** [comment]
Sorry, created new issue on plug-in repo: https://github.com/babel/eslint-plugin-babel/issues/27
